pyiceberg/io/pyarrow.py

```diff
@@ -1594,29 +1595,88 @@ def parquet_path_to_id_mapping(
     return result
 
 
-def fill_parquet_file_metadata(
```

Comment:
Removing this method will break daft: https://github.com/Eventual-Inc/Daft/pull/2016

However, I wouldn't say I like that we assign this in place, so maybe we should go for it.

Reply:
Awesome :)